waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -171,12 +168,13 @@ when defined(rlnzerokit):
     var
       keypair = MembershipKeyPair(idKey: secret, idCommitment: public)
 
-    return some(keypair)
+    return ok(keypair)
 
-proc createRLNInstance*(d: int = MerkleTreeDepth): RLNResult {.raises: [Defect, IOError].} =
+proc createRLNInstance*(d: int = MerkleTreeDepth): RLNResult =
```

Comment:
There seems to be no possibility for errors in this proc, right? then why result type for the output?

Reply:
When the rln instance creation fails, etc we do return an error as part of the RlnResult. This would allow appropriate error handling.